Git/config.sublime-syntax

```diff
@@ -0,0 +1,206 @@
+scope: source.gitconfig
+
+variables:
+  escape_char: \\"bnt
+  section_name: '[\w.-]+(?=[\s\]])'  # lookahead catches missing space, e.g. color"branch".
+  space_or_eol: (?=(?:\s|$))         # makes values on last line match properly
+  variable_name: '[a-zA-Z][\w-]*'
+  zero_to_255: 25[0-5]|2[0-4][0-9]|1\d\d|[1-9][0-9]|[0-9]
+
+contexts:
+  prototype:
+    - include: comments
+
+  main:
+    - include: section
+
+  boolean:
+    # 0 and 1 are omitted due to mismatches (e.g. "log -1", "HEAD~1"). They will
+    # be matched as unquoted strings instead.
+    - match: \b(?:yes|true|on|off|no|false){{space_or_eol}}
+      scope: constant.language.gitconfig
+
+  color:
+    - include: color-named      # red, blue, green
+    - include: color-attribute  # bold, italic, underline
+    - include: color-numeric    # 0-255
+
+  color-attribute:
+    - match: \b(?:no-?)?(?:ul|strike|reverse|italic|dim|bold|blink){{space_or_eol}}
+      scope: support.constant.color-attribute.gitconfig
+
+  color-named:
+    - match: \b(?:yellow|white|red|normal|magenta|green|cyan|blue|black|auto){{space_or_eol}}
+      scope: support.constant.color.gitconfig
+
+  color-numeric:
+    - match: '\b(?:{{zero_to_255}}){{space_or_eol}}'
+      scope: constant.other.color.rgb-value.gitconfig
+
+  comments:
+    - match: '[#;]'
+      scope: punctuation.definition.comment.gitconfig
+      push:
+        - meta_scope: comment.line.gitconfig
+        - match: $\n?  # consume newline so completions aren't shown at EOL when
+          pop: true    # typing comments
+
+  # The only valid escapes: '\b', '\n', '\t', '\"', '\\'.
+  escape:
+    - include: trailing-slash
+    - match: \\[^{{escape_char}}]
+      scope: invalid.illegal.escape.gitconfig
+    - match: \\[{{escape_char}}]
+      scope: constant.character.escape.gitconfig
+
+  illegal-newline:
+    - match: \n
+      scope: invalid.illegal.gitconfig
+
+  illegal-slash:
+    - match: \\
+      scope: invalid.illegal.gitconfig
+
+  # changed = red
+  # untracked = bold green
+  key-color-pair:
+    - match: ^\s*({{variable_name}})\s*(\=)
+      captures:
+        1: variable.other.readwrite.gitconfig
+        2: keyword.operator.assignment.gitconfig
+      push:
+        - meta_scope: meta.key-value-pair.gitconfig
+        - meta_content_scope: meta.value.gitconfig
+        - match: '(?=^\s*{{variable_name}}\s*\=)'  # next key-value pair
+          pop: true
+        - match: (?=^\s*\[)                        # start of new section
+          pop: true
+        - include: color
+
+  # key = val
+  key-value-pair:
+    - match: ^\s*({{variable_name}})\s*(\=)
+      captures:
+        1: variable.other.readwrite.gitconfig
+        2: keyword.operator.assignment.gitconfig
+      push:
+        - meta_scope: meta.key-value-pair.gitconfig
+        - meta_content_scope: meta.value.gitconfig
+        - match: '(?=^\s*{{variable_name}}\s*\=)'  # next key-value pair
+          pop: true
```

Comment:
This match+pop is a bit weird.
```ini
[x]
    a = b
        c
    d = e
    f =
```
```
~> git config --get x.a
b
~> git config --get x.d
e
~> git config --get x.c
   # (this prints a newline (empty string?))
~> git config --get x.asdf
~> # (prints nothing)
~> git config --get x.f
   # (this prints a newline (empty string?))
~>
```
It seems values really *end* when a newline is encountered. Except of course when we're in a quoted string. But the syntax now suggests that `c` is (part of) a *value* while it seems to be a key for an empty value according to git. So my suggestion is to pop once a new-line is encountered inside the `value` context and to remove this match+pop.

Reply:
This should be resolved by my latest push.